src/stores/PreferenceStore/PreferenceStore.ts

```diff
@@ -703,25 +728,24 @@ export class PreferenceStore {
 
         // set preference in variable
         if (key === PreferenceKeys.LOG_EVENT) {
-            if (!Event.isEventTypeValid(value)) {
+            if (!Event.isTypeValid(value)) {
                 return false;
             }
-            let eventList = this.preferences.get(PreferenceKeys.LOG_EVENT);
-            if (!eventList || !Array.isArray(eventList)) {
-                eventList = [];
-            }
-            if (eventList.includes(value)) {
-                eventList = eventList.filter(e => e !== value);
-            } else {
-                eventList.push(value);
-            }
+            let eventList = this.setCheckBoxList(this.preferences.get(PreferenceKeys.LOG_EVENT), value);
             this.preferences.set(PreferenceKeys.LOG_EVENT, eventList);
             return yield ApiService.Instance.setPreference(PreferenceKeys.LOG_EVENT, eventList);
+        } else if (key === PreferenceKeys.GLOBAL_AUTO_WCS_MATCHING) {
+            if (!WCSMatching.isTypeValid(value)) {
+                return false;
+            }
+            let bin = this.preferences.get(PreferenceKeys.GLOBAL_AUTO_WCS_MATCHING);
```

Comment:
`bin` could be confused with `binning`. I suggest we rename it to `binaryNumber`.
Use `const` if the variable won't be changed.

Reply:
okay